core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -744,7 +732,7 @@ private[spark] class BlockManager(
       level: StorageLevel,
       tellMaster: Boolean = true,
       effectiveStorageLevel: Option[StorageLevel] = None,
-      keepReadLock: Boolean = false): Option[PutResult] = {
+      keepReadLock: Boolean = false): Option[Option[Iterator[Any]]] = {
```

Comment:
This was discussed extensively on https://github.com/apache/spark/pull/11502 and will be fixed once I merge that PR.